java/compaction-job-creation/src/main/java/sleeper/compaction/strategy/impl/CompactionFactory.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.compaction.strategy.impl;
+
+import org.apache.commons.lang3.tuple.MutablePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sleeper.compaction.job.CompactionJob;
+import sleeper.configuration.properties.InstanceProperties;
+import sleeper.configuration.properties.table.TableProperties;
+import sleeper.statestore.FileInfo;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.FILE_SYSTEM;
+import static sleeper.configuration.properties.table.TableProperty.DATA_BUCKET;
+import static sleeper.configuration.properties.table.TableProperty.ITERATOR_CLASS_NAME;
+import static sleeper.configuration.properties.table.TableProperty.ITERATOR_CONFIG;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_NAME;
+
+public class CompactionFactory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CompactionFactory.class);
+
+    private final String tableName;
+    private final String tableBucket;
+    private final String fs;
+    private final String iteratorClassName;
+    private final String iteratorConfig;
+
+    public CompactionFactory(InstanceProperties instanceProperties, TableProperties tableProperties) {
+        tableName = tableProperties.get(TABLE_NAME);
+        tableBucket = tableProperties.get(DATA_BUCKET);
+        fs = instanceProperties.get(FILE_SYSTEM);
+        iteratorClassName = tableProperties.get(ITERATOR_CLASS_NAME);
+        iteratorConfig = tableProperties.get(ITERATOR_CONFIG);
+        LOGGER.info("Initialised CompactionFactory with table name {}, bucket for table data {}, fs {}",
+                tableName, tableBucket, fs);
```

Comment:
I think in all other parts of the codebase we use this in constructors, even if it's technically not necessary because the variables passed into the constructor don't have the same name.